Common/Orders/Fees/InteractiveBrokersTieredFeeModel.cs

```diff
@@ -0,0 +1,287 @@
+                    // FINRA Pass Through Fees.
+                    var passThroughFee = Math.Min(8.3m, feeResult * 0.00056m);
+
+                    feeResult += regulatoryFee + clearingFee + exchangeFee + passThroughFee;
+
+                    // Update the monthly volume shares traded
+                    _monthlyTradeVolume[SecurityType.Equity] += quantity;
+                    break;
+
+                case SecurityType.Cfd:
+                    InteractiveBrokersFeeHelper.CalculateCfdFee(security, order, out feeResult, out feeCurrency);
+                    break;
+                    
+                case SecurityType.Crypto:
+                    // Update the monthly value traded
+                    _monthlyTradeVolume[SecurityType.Crypto] += InteractiveBrokersFeeHelper.CalculateCryptoFee(security, order, _cryptoCommissionRate, CryptoMinimumOrderFee, out feeResult, out feeCurrency);
+                    break;
+
+                default:
+                    // unsupported security type
+                    throw new ArgumentException(Messages.FeeModel.UnsupportedSecurityType(security));
+            }
+
+            _lastOrderTime = order.Time;
+
+            return new OrderFee(new CashAmount(
+                feeResult,
+                feeCurrency));
+        }
+
+        private CashAmount UnitedStatesFutureFees(Security security)
+        {
+            IDictionary<string, decimal[]> fees;
+            IDictionary<string, decimal> exchangeFees;
+            decimal[] ibFeePerContract;
+            decimal exchangeFeePerContract;
+            string symbol;
+
+            switch (security.Symbol.SecurityType)
+            {
+                case SecurityType.Future:
+                    fees = _usaFuturesFees;
+                    exchangeFees = InteractiveBrokersFeeHelper.UsaFuturesExchangeFees;
+                    symbol = security.Symbol.ID.Symbol;
+                    break;
+                case SecurityType.FutureOption:
+                    fees = _usaFutureOptionsFees;
+                    exchangeFees = InteractiveBrokersFeeHelper.UsaFutureOptionsExchangeFees;
+                    symbol = security.Symbol.Underlying.ID.Symbol;
+                    break;
+                default:
+                    throw new ArgumentException(Messages.InteractiveBrokersFeeModel.UnitedStatesFutureFeesUnsupportedSecurityType(security));
+            }
+
+            if (!fees.TryGetValue(symbol, out ibFeePerContract))
+            {
+                ibFeePerContract = new[] { 0.85m, 0.65m, 0.45m, 0.25m };
+            }
+
+            if (!exchangeFees.TryGetValue(symbol, out exchangeFeePerContract))
+            {
+                exchangeFeePerContract = 1.60m;
+            }
+
+            // Add exchange fees + IBKR regulatory fee (0.02)
+            return new CashAmount(ibFeePerContract[_futureCommissionTier] + exchangeFeePerContract + 0.02m, Currencies.USD);
```

Comment:
This implementation seems like a slightly more generic version of the one at `InteractiveBrokersFeeModel` seems they can be mostly shared, where `InteractiveBrokersFeeModel` just uses tier 0?